app/Nova/Actions/ActivateKiosk.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Nova\Actions;
+
+use Illuminate\Support\Collection;
+use Laravel\Nova\Actions\Action;
+use Laravel\Nova\Fields\ActionFields;
+use Laravel\Nova\Fields\Heading;
+use Laravel\Nova\Fields\Text;
+
+class ActivateKiosk extends Action
+{
+    /**
+     * Perform the action on the given models.
+     *
+     * @param \Laravel\Nova\Fields\ActionFields $fields
+     * @param \Illuminate\Support\Collection $models
+     *
+     * @return array|string
+     */
+    public function handle(ActionFields $fields, Collection $models)
```

Comment:
This should be [`onlyOnDetail`](https://nova.laravel.com/docs/1.0/actions/defining-actions.html#action-visibility)